### ISSUE TITLE
API to get the attribute name from value? 

### ISSUE THREAD

**Issue by jaseemabid** [bigjason/django-choices#47]
Consider the example from the README.

```
class PersonType(DjangoChoices):
    customer = ChoiceItem("C", "Cust label") 
    employee = ChoiceItem("E")
    groundhog = ChoiceItem("G")
```

Is there an API to get the attribute name as a string (`customer`) from the value (`C`) ?

The existing API `PersonType.labels` maps attributes to label, `PersonType.values` maps values to label, but I need the reverse. I need a map from value to attribute name.

**Comment by ashwch** [comment]
@jaseemabid You could do: `{v.value: k for k, v in PersonType._fields.items()}`

**Comment by sergei-maertens** [comment]
The suggestion from @ashwch is probably the most accurate. It's private API, though not expected to change anytime soon.

I have had similar needs, and will consider making it public API. However, this will add extra validation that each value *must* be unique. I'm not sure how the choices/field behave right now if they aren't.

If they are not unique, you risk having one attribute overwrite the other in the reverse mapping.

Thoughts?

**Comment by jaseemabid** [comment]
@sergei-maertens Yes, a public API would be nice. 

I'd raise an exception that values are not unique rather than silently return wrong data; which makes it usable on classes with unique values.

**Comment by sergei-maertens** [comment]
An exception is indeed the way to go, although I'm a bit worried about existing projects that may have duplicate values. This should probably go through a deprecation period first.

Silently *anything* is not an option, so no worries there.

**Comment by sergei-maertens** [comment]
@jaseemabid better late than never... this is now finally released on PyPI with version 1.5.1
